app/src/main/java/org/schabi/newpipe/fragments/detail/VideoDetailFragment.java

```diff
@@ -258,26 +383,39 @@ public void onResume() {
             updateFlags = 0;
         }
 
-        // Check if it was loading when the fragment was stopped/paused,
-        if (wasLoading.getAndSet(false)) {
-            selectAndLoadVideo(serviceId, url, name);
-        } else if (currentInfo != null) {
-            updateProgressInfo(currentInfo);
-        }
+        // Check if it was loading when the fragment was stopped/paused
+        if (wasLoading.getAndSet(false) && !wasCleared())
+            startLoading(false);
+    }
+
+    @Override
+    public void onStop() {
+        super.onStop();
+
+        if (!activity.isChangingConfigurations())
+            activity.sendBroadcast(new Intent(ACTION_VIDEO_FRAGMENT_STOPPED));
     }
 
     @Override
     public void onDestroy() {
         super.onDestroy();
+
+        // Stop the service when user leaves the app with double back press if video player is selected. Otherwise unbind
+        if (activity.isFinishing() && player != null && player.videoPlayerSelected()) stopService();
+        else unbind();
+
         PreferenceManager.getDefaultSharedPreferences(activity)
                 .unregisterOnSharedPreferenceChangeListener(this);
+        activity.unregisterReceiver(broadcastReceiver);
+        activity.getContentResolver().unregisterContentObserver(settingsContentObserver);
```

Comment:
Shouldn't you also set `broadcastReceiver` and `settingsContentObserver` to `null`, to prevent memory leaks?

Reply:
onDestroy() calls garbage collection once it ends so all instances will be collected and removed from heap. Setting them to null is not needed and unregistering from events is enough in this case